core/util/extendConfig.js

```diff
@@ -59,17 +62,25 @@ function htmlReport(config, userConfig) {
   config.compareReportURL = path.join(config.html_report, 'index.html');
 }
 
-function comparePaths(config) {
+function comparePaths (config) {
   config.comparePath = path.join(config.backstop, 'compare');
   config.tempCompareConfigFileName = temp.path({suffix: '.json'});
 }
 
-function captureConfigPaths(config) {
-  config.captureConfigFileName = path.join(config.backstop, 'capture', 'config.json');
+function captureConfigPaths (config) {
+  var captureDir = path.join(tmpdir, 'capture');
+  if (!fs.existsSync(captureDir)) {
+    fs.mkdirSync(captureDir);
+  }
+  var tempCompareConfigFileName = config.tempCompareConfigFileName;
+  config.tempCompareConfigFileName = '';
+  var configHash = hash(config);
+  config.tempCompareConfigFileName = tempCompareConfigFileName;
```

Comment:
Okey, I have checked this and remembered why I do like this.

This is an intended behavior for preventing different hashes because of a dynamic temp value. 

The hash value only needs to be change when you make a change in your backstop.json file. 

If we do this as your suggestion every time we are going to get different hash name and it will create a lot of tmp files (even backstop.json file is same) for each execution.